src/JSONdata/handle_json.rb

```diff
@@ -0,0 +1,117 @@
+require 'json'
+
+module FileHandler
+  # WRITER
+  def write_json(array, file_path)
+    opts = {
+      array_nl: "\n",
+      object_nl: "\n",
+      indent: '  ',
+      space_before: ' ',
+      space: ' '
+    }
+    File.write(file_path, JSON.generate(array, opts))
+  end
+
+  # READER
+  def read_json(file_path)
+    return unless File.exist?(file_path)
+
+    file = File.read(file_path)
+    JSON.parse(file)
+  end
+
+  # STORE BOOK IN JSON
+  def store_books
+    array = []
+    @list_books.each do |book|
+      array.push(
+        {
+          title: book.title,
+          author: book.author
+        }
+      )
+    end
+    write_json(array, 'JSONdata/books.json')
+  end
+
+  # READ STORED BOOKS
+  def read_books
+    parse_file = read_json('JSONdata/books.json')
+    parse_file.map do |book|
+      @list_books.push(Book.new(book['title'], book['author']))
+    end
+  end
+
+  # STORE PEOPLE
+  def store_people
+    array = []
+    @list_people.map do |person|
+      if person.instance_of?(Student)
+        array.push(
+          {
+            id: person.id,
+            type: person.class,
+            age: person.age,
+            name: person.name,
+            parent_permission: person.parent_permission
+          }
+        )
+      else
+        array.push(
+          {
+            id: person.id,
+            type: person.class,
+            age: person.age,
+            name: person.name,
+            specialization: person.specialization
+          }
+        )
+      end
+    end
+    write_json(array, 'JSONdata/people.json')
+  end
+
+  # READ PEOPLE
+  def read_people
+    parse_file = read_json('JSONdata/people.json')
+    parse_file.map do |person|
+      if person['type'] == 'Teacher'
+        teacher = Teacher.new(person['age'], person['name'], person['specialization'])
+        teacher.id = person['id']
+        @list_people.push(teacher)
+      else
```

Comment:
- [ ] Kindly make sure when you create a person, book or rental it should preserve the data in the respective JSON file as explained in the loom video [here](https://www.loom.com/share/c8e59a4aa0204724bad9635d74903b5f).

Reply:
hey @sja-thedude actually the requirements states that 

- On exit from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly (e.g. rentals.json)

so data are saved on exit not instantly when entered